### PR TITLE
Add ADR for autoupdate updates

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Documents the meaningful launcher-side updates to our autoupdate process.

### CODE REVIEW

File: docs/architecture/2023-10-17_autoupdate_v3.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+- Rebecca Mahany-Horton ([@RebeccaMahany](https://github.com/RebeccaMahany))
+
+## Status
+
+Accepted: October 2023
+
+Changes rolled out to nightly channel in October 2023. Rollout to beta and
+stable is still TBD.
+
+Supersedes: [Launcher Auto Update Process Version 2](2019-09-05_autoupdate_v2.md)
```
Comment (directionless) [inline docs/architecture/2023-10-17_autoupdate_v3.md:14]: It's worth dropping a note into that one's status linking here

File: docs/architecture/2023-10-17_autoupdate_v3.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+Third, we would like the location of updates to be configurable.
+
+See [Begin work on Autoupdate: The Next Generation](https://github.com/kolide/launcher/issues/954)
+for more details.
+
+## Decision
+
+We created a new update directory that lives, by default, in the root
+directory, but its location is configurable via flag.
+
```
Comment (directionless) [inline docs/architecture/2023-10-17_autoupdate_v3.md:38]: Probably need a sentence or paragraph here saying something like:

"We will use the go-tuf implementation of TUF to track our releases and what versions are in each channel."

**Review by James-Pickett — APPROVED** [review]

**Review by zackattack01 — APPROVED** [review]
this is really helpful, thank you!

**Review by directionless — APPROVED** [review]
Improvements!

**Comment by RebeccaMahany** [comment]
Relates to https://github.com/kolide/launcher/issues/954

**Pull request merged**